src/_pytest/cacheprovider.py

```diff
@@ -203,17 +203,33 @@ def set(self, key: str, value: object) -> None:
             with f:
                 f.write(data)
 
-    def _ensure_supporting_files(self) -> None:
-        """Create supporting files in the cache dir that are not really part of the cache."""
-        readme_path = self._cachedir / "README.md"
-        readme_path.write_text(README_CONTENT, encoding="UTF-8")
-
-        gitignore_path = self._cachedir.joinpath(".gitignore")
-        msg = "# Created by pytest automatically.\n*\n"
-        gitignore_path.write_text(msg, encoding="UTF-8")
+    def _ensure_cache_dir_and_supporting_files(self) -> None:
+        """Create the cache dir and its supporting files."""
+        if self._cachedir.is_dir():
+            return
 
-        cachedir_tag_path = self._cachedir.joinpath("CACHEDIR.TAG")
-        cachedir_tag_path.write_bytes(CACHEDIR_TAG_CONTENT)
+        with tempfile.TemporaryDirectory() as newpath:
```

Comment:
as i just took note @bluetech already explicitly mentioned we need the TemporaryDirectory in the same directory the cache will be in 

as such the dir parameter would need to be the parent directory and the prefix should be something that hints at the intent

Reply:
Please use existing discussions rather than starting new ones. As I said [here](https://github.com/pytest-dev/pytest/pull/12168#discussion_r1547161910):
> I don't think we should do that - we may end up with that directory lingering in the interrupt case. I don't know enough about how signal handlers work in python.

If you create a temporary directory in the same directory as the cache you may end up leaving the temporary directory there if you are interrupted.